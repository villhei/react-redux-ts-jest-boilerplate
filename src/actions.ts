
import { Action as ReduxAction } from 'redux'


export interface Action<T> extends ReduxAction {
  readonly type: string,
  readonly payload: T
}

export const SET_COUNT = 'SET_COUNT'


export function setCountAction(n: number): Action<number> {
  return {
    type: SET_COUNT,
    payload: n
  }
}

export function isSetCountAction(candidate: any): candidate is Action<number> {
  return candidate.type === SET_COUNT
}