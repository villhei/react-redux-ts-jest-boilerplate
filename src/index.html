<html>

<body>
  <div id="app"></div>
</body>

<script async src="./index.tsx"></script>

</html>