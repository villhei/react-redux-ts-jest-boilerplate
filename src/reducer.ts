import { Action } from 'redux'
import { isSetCountAction } from './actions'

export type State = {
  count: number
}

export const initialState: State = {
  count: 0
}

export default (state = initialState, action: Action) => {
  if (isSetCountAction(action)) {
    return {
      ...state,
      count: action.payload
    }
  }
  return state
}