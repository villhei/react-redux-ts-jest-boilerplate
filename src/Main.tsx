import * as React from 'react'
import { connect, Dispatch } from 'react-redux'
import { State } from './reducer'
import { setCountAction } from './actions'

export type Props = {
  count: number,
  setCount: (n: number) => void
}

export const Main = (props: Props) => {
  const { setCount, count } = props
  return (
    <div>
      <h1>Hello there, the count is {count}</h1>
      <button onClick={() => setCount(props.count + 1)}>
        Increment
      </button>
    </div>
  )
}

const mapStateToProps = (state: State): State => state

const mapDispatchToProps = (dispatch: Dispatch<State>) => ({
  setCount: (n: number) => { dispatch(setCountAction(n)) }
})

export default connect(mapStateToProps, mapDispatchToProps)(Main)