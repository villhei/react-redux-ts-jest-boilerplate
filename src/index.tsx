import * as React from 'react'
import * as ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import Main from './Main'
import store from './store'

const root = document.querySelector('#app')

const App = () => <h1>Slide </h1>

if (root) {
  ReactDOM.render((
    <Provider store={store}>
      <Main />
    </Provider>
  ), root)
} else {
  console.log('Failed to find the root node')
}