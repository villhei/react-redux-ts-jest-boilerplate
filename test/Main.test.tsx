import * as React from 'react'
import { Main } from '../src/Main'
import { shallow } from 'enzyme'

describe('Main container', () => {
  it('should render the component', () => {
    const props = {
      count: 0,
      setCount: jest.fn()
    }
    const element = shallow(<Main {...props} />)
    expect(element).toMatchSnapshot()
  })
})
