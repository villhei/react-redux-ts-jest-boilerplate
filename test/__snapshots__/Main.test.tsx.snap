// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main container should render the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    count={0}
    setCount={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Hello there, the count is 
          0
        </h1>,
        <button
          onClick={[Function]}
        >
          Increment
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hello there, the count is ",
            0,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Hello there, the count is ",
          0,
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Increment",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Increment",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Hello there, the count is 
            0
          </h1>,
          <button
            onClick={[Function]}
          >
            Increment
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hello there, the count is ",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Hello there, the count is ",
            0,
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Increment",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Increment",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
